tests/testHtmIndex.py

```diff
@@ -0,0 +1,213 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import os
+import tempfile
+import shutil
+import unittest
+import string
+from collections import Counter
+
+import numpy
+
+import lsst.utils
+import lsst.afw.table as afwTable
+import lsst.afw.geom as afwGeom
+import lsst.afw.coord as afwCoord
+import lsst.daf.persistence as dafPersist
+from lsst.pipe.tasks.htmIndexEngine import IngestIndexedReferenceTask
+from lsst.pipe.tasks.loadIndexedReferenceObjects import LoadIndexedReferenceObjectsTask
+
+obs_test_dir = lsst.utils.getPackageDir('obs_test')
+input_dir = os.path.join(obs_test_dir, "data", "input")
+
+def make_coord(ra, dec):
+    return afwCoord.IcrsCoord(afwGeom.Angle(ra, afwGeom.degrees), afwGeom.Angle(dec, afwGeom.degrees))
+
+class HtmIndexTestCase(lsst.utils.tests.TestCase):
+    @staticmethod
+    def make_sky_catalog(out_path, size=1000):
+        numpy.random.seed(123)
+        ident = numpy.arange(1, size+1, dtype=int)
+        ra = numpy.random.random(size)*360.
+        dec = numpy.degrees(numpy.arccos(2.*numpy.random.random(size) - 1.))
+        dec -= 90.
+        a_mag = 16. + numpy.random.random(size)*4.
+        a_mag_err = 0.01 + numpy.random.random(size)*0.2
+        b_mag = 17. + numpy.random.random(size)*5.
+        b_mag_err = 0.02 + numpy.random.random(size)*0.3
+        is_photometric = numpy.random.randint(2, size=size)
+        is_resolved = numpy.random.randint(2, size=size)
+        is_variable = numpy.random.randint(2, size=size)
+        extra_col1 = numpy.random.normal(size=size)
+        extra_col2 = numpy.random.normal(1000., 100., size=size)
+        def get_word(word_len):
+            return "".join(numpy.random.choice([s for s in string.ascii_letters], word_len))
+        extra_col3 = numpy.array([get_word(num) for num in numpy.random.randint(11, size=size)])
+
+        dtype = numpy.dtype([('id', float), ('ra_icrs', float), ('dec_icrs', float), ('a', float),
+                             ('a_err', float), ('b', float), ('b_err', float), ('is_phot', int),
+                             ('is_res', int), ('is_var', int), ('val1', float), ('val2', float),
+                             ('val3', '|S11')])
+
+        arr = numpy.array(zip(ident, ra, dec, a_mag, a_mag_err, b_mag, b_mag_err, is_photometric, is_resolved,
+                           is_variable, extra_col1, extra_col2, extra_col3), dtype=dtype)
+        numpy.savetxt(out_path+"/ref.txt", arr, delimiter=",",
```

Comment:
I think this demonstrates my concern about having the ingestion read CSV rather than FITS.  Converting FITS catalogs covering 3pi steradians to 21 mag to CSV is going to be very slow and consume a lot of disk space.